src/client/Client.cc

```diff
@@ -11266,18 +11278,6 @@ int64_t Client::_read(Fh *f, int64_t offset, uint64_t size, bufferlist *bl,
     // is duplicated and modified and exists in
     // C_Read_Sync_NonBlocking::finish().
 
-    // trim read based on file size?
```

Comment:
I'm trying to keep things consistent. Right now case 1 (`_read_async`) and case 3 (`_read_sync`) both check again if the size/len is 0 within the functions. They both are only called from `_read()`. Do we want to remove those checks, or add a check to `C_Read_Sync_NonBlocking` to complete immediately in `start()` if `len == 0`? That way we can potentially avoid bugs later on.

Reply:
> Right now case 1 (_read_async) and case 3 (_read_sync) both check again if the size/len is 0 within the functions. They both are only called from _read()

I did thought about this initially but then wondered - what if in future these member functions are called independently from some new API? I felt that considering they are helper functions to carry out async/sync read; i kept those functions intact. WDYT?